deepinv/models/icnn.py

```diff
@@ -0,0 +1,146 @@
+# Code borrowed from https://github.com/ZhenghanFang/learned-proximal-networks
+import numpy as np
+import torch
+from torch import nn
+
+
+class ICNN(nn.Module):
+    r"""
+    Input Convex Neural Network (ICNN) with convolutional layers.
```

Comment:
Add space for having a nice title in the doc
```suggestion
    Input Convex Neural Network (ICNN).
    
```